client/src/reducers/index.js

```diff
@@ -16,6 +16,7 @@ import layoutChoice from "./layoutChoice";
 import controls from "./controls";
 import annotations from "./annotations";
 import genesets from "./genesets";
+import genesetsUI from "./genesetsUI";
```

Comment:
I noticed you did not add genesetsUI to the undoable state list.  Is that intentional?  

I suggest confirming that undo/redo works correctly with genesets, as I suspect it is broken without this.

Reply:
You cannot click undo and redo while the modals are popped, and this reducer only controls the `show` and `cancel/hide` for the modals

Reply:
gotcha - so as long as undo/redo is confirmed to otherwise work, it sounds like all is good....

Reply:
I believe you are correct @bkmartinjr, it doesn't break it completely (I didn't notice before) but if you undo/redo past the point of popping the modal, there is an error in console. Do you repro?

Reply:
I don't reproduce, but I also am unclear on the sequence of steps you performed to see the error. Can you give me a step-by-step example?

In principle, leaving `genesetsUI` off the undoable state just means that it will be untouched by undo/redo.  So if the reducer _only_ captures state while the modal is active, AND the modal and undo/redo UI are exclusive, I don't think you need to list it in the undo config.   That said, I may be misunderstanding the reducer state. If you send me the test case, I can dig future.

Last thought - also include the error that you see, on the off chance that it has nothing to do with undo/redo.

Reply:
I saw a 404 on put, which has been resolved thanks to @bkmartinjr